src/AutoRest.CSharp/Common/Output/Expressions/Statements/DeclareLocalFunctionStatement.cs

```diff
@@ -9,5 +9,12 @@
 
 namespace AutoRest.CSharp.Common.Output.Expressions.Statements
 {
-    internal record DeclareLocalFunctionStatement(CodeWriterDeclaration Name, IReadOnlyList<Parameter> Parameters, CSharpType ReturnType, ValueExpression Body) : DeclarationStatement;
+    internal record DeclareLocalFunctionStatement(CodeWriterDeclaration Name, IReadOnlyList<Parameter> Parameters, CSharpType ReturnType, ValueExpression? BodyExpression, MethodBodyStatement? BodyStatement) : DeclarationStatement
```

Comment:
this syntax also creates a ctor that could take both `BodyExpression` and `BodyStatement`, we should not have this ctor.

Reply:
The idea is one or the other will not be null

Reply:
Well this ctor never prevent the user to give values to both statement and body. The ctors of this type should look like this:
```csharp
public DeclarreLocalFunctionStatement(Name, Parameters, ReturnType, BodyExpression) : this(Name, Parameters, ReturnType, BodyExpression, null)
public DeclarreLocalFunctionStatement(Name, Parameters, ReturnType, BodyStatement) : this(Name, Parameters, ReturnType, null, BodyStatement)
private DeclarreLocalFunctionStatement(Name, Parameters, ReturnType, BodyExpression?, BodyStatement?)
```
We did this in `Method` class, therefore I believe we should do the same here.

Reply:
I don't think this works on a record.  Once we switch to classes we can do this but I don't want to take that change in this PR.